tests/test-customize-changesets-controller.php

```diff
@@ -99,29 +518,1239 @@ public function test_update_item() {
+		$this->assertNotSame( $data['next_changeset_uuid'], $uuid_before );
+	}
+
+	/**
+	 * Test that publishing an existing changeset with update_item() returns a new changeset UUID.
+	 */
+	public function test_update_item_edit_has_next_changeset_id_after_publish() {
+		wp_set_current_user( self::$admin_id );
+
+		$uuid_before = wp_generate_uuid4();
+
+		$manager = new WP_Customize_Manager( array(
+			'changeset_uuid' => $uuid_before,
+		) );
+		$manager->save_changeset_post();
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $uuid_before ) );
+		$request->set_body_params( array(
+			'status' => 'publish',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$data = $response->get_data();
+		$this->assertTrue( isset( $data['next_changeset_uuid'] ) );
+		$this->assertNotSame( $data['next_changeset_uuid'], $uuid_before );
+	}
+
+	/**
+	 * Test that creating a published changeset with update_item() updates a valid setting.
+	 */
+	public function test_update_item_create_with_bloginfo() {
+		wp_set_current_user( self::$admin_id );
+
+		$uuid = wp_generate_uuid4();
+		$blogname_after = rand_str();
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $uuid ) );
+		$request->set_body_params( array(
+			'customize_changeset_data' => array(
+				'blogname' => array(
+					'value' => $blogname_after,
+				),
+			),
+			'status' => 'publish',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertSame( 200, $response->get_status() );
+		$this->assertSame( $blogname_after, get_option( 'blogname' ) );
+	}
+
+	/**
+	 * Test that publishing a changeset with update_item() updates a valid setting.
+	 */
+	public function test_update_item_edit_with_bloginfo() {
+		wp_set_current_user( self::$admin_id );
+
+		$blogname_after = rand_str();
+
+		$manager = new WP_Customize_Manager();
+		$manager->save_changeset_post( array(
+			'data' => array(
+				'blogname' => array(
+					'value' => $blogname_after,
+				),
+			),
+		) );
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $manager->changeset_uuid() ) );
+		$request->set_body_params( array(
+			'status' => 'publish',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertSame( 200, $response->get_status() );
+		$this->assertSame( $blogname_after, get_option( 'blogname' ) );
+	}
+
+	/**
+	 * Test that update_item() returns setting validities.
+	 */
+	public function test_update_item_setting_validities() {
+		wp_set_current_user( self::$admin_id );
+
+		$bad_setting = rand_str();
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', wp_generate_uuid4() ) );
+		$request->set_body_params( array(
+			'customize_changeset_data' => array(
+				$bad_setting => array(
+					'value' => 'Foo',
+				),
+			),
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertSame( 200, $response->get_status() );
```

Comment:
This is another interesting one. I think the REST API should actually fail this entirely, and the `setting_validities` should be part of the error response.